nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader, Packet  # type: ignore
+from typing import Callable, List, Optional
+
+
+def generate_histogram(data: list[int], buckets: int, bucket_size: int) -> List[int]:
+    if len(data) == 0:
+        return []
```

Comment:
You gave me some ideas, so I made thse changes:
- simplified the code around `sorted_values` and eliminated that unecessary variable
- moved `len()==0` check into the call site. It will still throw ValueError in the `max()` function however the call site will protect against that which is expected since there might be no data captured or the filter was too harsh